syntax="proto3";

/**
This will be a work in progress for all of the 
*/
service ListeningComms
{
    rpc ClockData (GenericRequest) returns (ClockResponse);

    rpc DateData (GenericRequest) returns (DateResponse);

    rpc EmailData (GenericRequest) returns (stream EmailResponse);

    rpc CalendarData (GenericRequest) returns ( stream CalendarResponse);

    rpc WeatherData (GenericRequest) returns (stream WeatherResponse);

    rpc NewsData (GenericRequest) returns (stream NewsResponse);
}

service ConnectionComms
{
    rpc IntialConnection (ConnectionRequest) returns (Response);
    rpc KeepAlive (Request) returns (Response);
    rpc DisconnectNode (Request) returns (Response);
}

message ConnectionRequest
{
    int32 connect = 1;
    string nodeHostIP = 2;
}

message DisconnectRequest
{
    int32 disconnect = 1;
    string nodeHostIP = 2;
}

message ConnectionStatus
{
    int32 alive = 1;
}

message ConnectionStatusResponse
{
    int32 stillAlive = 1;
}



message GenericRequest
{
    string requestData = 1;
}


message ClockResponse
{
    int32 hour = 1;
    int32 minute = 2;
    int32 second = 3;

    int32 day = 4;
    int32 month = 5;
    int32 year = 6;
}

message DateResponse
{
    int32 day = 1;
    int32 month = 2;
    int32 year = 3;
}

message EmailResponse
{
    string email = 1;
}

message CalendarResponse
{
    string events = 1;
}

message WeatherResponse
{
    string data = 1;
}

message NewsResponse
{
    string headline = 1;
}

